test/test_functionalization.py

```diff
@@ -1542,6 +1542,35 @@ def forward(self, x_1):
     add = torch.ops.aten.add.Tensor(view_1, 1);  view_1 = None
     return add""")
 
+    def test_python_functionalization_zero_tensor(self):
```

Comment:
We should probably also run these tests, but with C++ functionalization, to make sure the behavior is the same.

NB: Despite DispatchKey::ZeroTensor being ABOVE DispatchKey::Functionalize, what actually happens for C++ functionalization is that DispatchKey::Functionalize runs first! (Because we use Functionalize as a Mode, and it ends up wrapping the constructed EfficientZeroTensor) This is OK because one cannot do any in-place operations to EfficientZeroTensor and is what we want to happen with Python FunctionalTensor as well.

Reply:
This is definitely fair, good call out. I'll add a test for both.